include/ddsrouter/event/impl/SignalHandler.ipp

```diff
@@ -131,20 +146,58 @@ void SignalHandler<SigNum>::signal_handler_routine_(
     }
 }
 
+template <int SigNum>
+void SignalHandler<SigNum>::signal_handler_thread_routine_() noexcept
+{
+    signal(SigNum, [](int)
+            {
+                signal_received_.store(true);
```

Comment:
There is a small problem here. Every change of variables inside a condition variable predicate must be guarded by a mutex (else it could lead to a lost notify).

However, this is very unlikely, and there should not be mutex locks inside the signal function.
Thus, I think it is ok. However I would add it as a comment, it may save much time in the future for someone to debug a possible bug.